### DISCUSSION TITLE
Problem: Unable to search Issues, PRs and Discussions at the same time.

### DISCUSSION THREAD

**Discussion by rszyma** [community/community#74821] [Discussions]
### Select Topic Area

Product Feedback

### Body

Currently, Github UI allows to search both Issues and PRs at the same time by going to either Issues or PRs tab and removing `is:issue` or `is:pr` search filter. But removing `is` filter DOESN'T search discussions. 

It's a problem, because before posting a new issue/discussion I have to make sure that I won't create a duplicate, and depending on whether/how Discussions are utilized in a particular project, it means that most of the time I need to search for duplicates twice! First time in Issues tab and second time in Discussions tab.

A proposal:
1. Add `is:discussion` search filter
2. Use `is:discussion` as a default search filter to search box in Discussions 
3. Allow searching Discussions from Issues or Pull Request Search (and vice versa, allow searching issues/PRs by removing `is:discussion` from search query)

**Comment by Unmesh100** [comment]
Your feedback highlights an important usability issue within GitHub's search functionality. By suggesting the addition of an is:discussion search filter and the ability to search across Issues, Pull Requests, and Discussions simultaneously, you're proposing a valuable improvement to streamline the process of avoiding duplicate posts.

Your proposal includes:

Adding is:discussion search filter – This would allow users to easily filter for Discussions within a project.
Using is:discussion as the default filter for Discussions – This would ensure that searches within the Discussions tab start with relevant results.
Unified Search Across All Tabs – Allowing users to remove search filters (is:issue, is:pr, is:discussion) to search all relevant content in one go, reducing redundancy and time spent verifying duplicates across tabs.
This would indeed make the search process more efficient, especially for contributors who are cautious about not duplicating issues or discussions. Have you considered sharing this directly on GitHub's feedback channels or community forums? Your suggestion could gain traction from other users experiencing the same pain points.
